### PR TITLE
Refactoring RDAT parser to read Smith Chart report

### PR DESCRIPTION
**Pull request by Samuelopez-ansys**
If Smith Chart is detected, the output dictionary will have the info from the real and imag. part like 2 different traces

### CODE REVIEW

File: pyaedt/generic/report_file_parser.py [hunk 1]
```diff
@@ -17,25 +17,53 @@ def parse_rdat_file(file_path):
     for report_name in report_data:
         report_dict[report_name] = {}
         for trace, trace_data in report_data[report_name]["Traces"].items():
-            x_data = trace_data["TraceComponents"]["TraceDataComps"]["0"]
-            y_data = trace_data["TraceComponents"]["TraceDataComps"]["1"]
-            all_x_values = x_data["TraceDataCol"]["ColumnValues"]
-            all_y_values = y_data["TraceDataCol"]["ColumnValues"]
+            if trace_data["TraceComponents"]["TraceDataComps"]["0"]["TraceDataCol"]["ParameterType"] == "ComplexParam":
+                all_data = trace_data["TraceComponents"]["TraceDataComps"]["0"]
```
Comment (beliaev-maksim) [inline pyaedt/generic/report_file_parser.py:21]: cannot this be out of `if` block ?

File: pyaedt/generic/report_file_parser.py [hunk 2]
```diff
@@ -17,25 +17,53 @@ def parse_rdat_file(file_path):
...
-                "x_name": x_data["TraceCompExpr"],
-                "x_unit": si_unit_x,
-                "y_unit": si_unit_y,
-                "curves": {},
-            }
-            for curve, curve_data in trace_data["CurvesInfo"].items():
-                report_dict[report_name][trace_data["TraceName"]]["curves"][curve_data[1]] = {
-                    "x_data": all_x_values[0 : curve_data[0]],
-                    "y_data": all_y_values[0 : curve_data[0]],
+                si_unit_x = SI_UNITS[unit_system(x_data["TraceDataCol"]["Units"])]
```
Comment (beliaev-maksim) [inline pyaedt/generic/report_file_parser.py:53]: can this be out of `if` ?

File: pyaedt/generic/report_file_parser.py (hunk 1 above)
Comment (Samuelopez-ansys) [reply pyaedt/generic/report_file_parser.py:21]: This one, you are are right

File: pyaedt/generic/report_file_parser.py (hunk 2 above)
Comment (Samuelopez-ansys) [reply pyaedt/generic/report_file_parser.py:53]: This one, for Smith Chart, the key is different ("PrimarySweepCol")

**Review by maxcapodi78 — APPROVED** [review]

**Review by beliaev-maksim — APPROVED** [review]

**Pull request merged**